mesonpy/__init__.py

```diff
@@ -685,6 +686,44 @@ def _strings(value: Any, name: str) -> List[str]:
     return scheme(table, 'tool.meson-python')
 
 
+def _validate_config_settings(config_settings: Dict[str, Any]) -> Dict[str, Any]:
+    """Validate options received from build frontend."""
+
+    def _string(value: Any, name: str) -> str:
+        if not isinstance(value, str):
+            raise ConfigError(f'only one value for "{name}" can be specified')
+        return value
+
+    def _bool(value: Any, name: str) -> bool:
+        return True
+
+    def _string_or_strings(value: Any, name: str) -> List[str]:
+        return list([value,] if isinstance(value, str) else value)
+
+    options = {
+        'builddir': _string,
+        'editable-verbose': _bool,
+        'dist-args': _string_or_strings,
+        'setup-args': _string_or_strings,
+        'compile-args': _string_or_strings,
+        'install-args': _string_or_strings,
```

Comment:
We should not allow strings here.

Reply:
Please make up your mind. A little while ago you said that we need to allow frontends to pass single arguments as strings and that I broke that while refactoring.

Reply:
Sorry, sorry, I forgot this code was for `config_settings`.